integration-provisioning/src/main/java/org/symphonyoss/integration/provisioning/service/UserService.java

```diff
@@ -112,7 +113,7 @@ public void setupBotUser(IntegrationSettings settings, Application app) {
       String message = logMessage.getMessage(USER_NOT_FOUND_MESSAGE, userId.toString());
       throw new UserSearchException(message);
     } else {
-      updateUser(sessionToken, user, name, avatar);
+      updateUser(sessionToken, user, name, avatar, emailAddress);
```

Comment:
You might encapsulate the logic to get email address inside the method getEmail().

Something like that:
```
  private String getEmail(Application app, String username) {
    String emailAddress = certificateService.getEmailAddressFromApplicationCertificate(app);

    if (StringUtils.isNotEmpty(emailAddress)) {
      return emailAddress;
    }

    Matcher matcher = VALID_EMAIL_ADDRESS_REGEX.matcher(username);
    if (matcher.matches()) {
      return username;
    }

    return username + EMAIL_DOMAIN;
  }
```